File: Java/decorator/Granulado.java
package deco;

public class Granulado extends SorveteDecorator{
	public Granulado(ISorvete objetoDecorado) {
		super(objetoDecorado);
	}

	@Override
	public float preco() {
		return this.objetoDecorado.preco() + 2.5f;
	}
}

File: Decorator/Granulado.php
<?php
require_once "./ISorvete.php";

class Granulado implements ISorvete{
    private $objetoDecorado;

    public function __construct(ISorvete $objetoDecorado)
    {
        $this->objetoDecorado = $objetoDecorado;
    }

    public function preco(){
        return $this->objetoDecorado->preco() + 2.5;
    }
}